index.js

```diff
@@ -72,22 +47,29 @@ task(TASK_COMPILE, async function (args, hre, runSuper) {
 
   const outputDirectory = path.resolve(hre.config.paths.root, config.path);
 
-  if (!outputDirectory.startsWith(hre.config.paths.root)) {
-    throw new HardhatPluginError('resolved path must be inside of project directory');
-  }
-
   if (outputDirectory === hre.config.paths.root) {
     throw new HardhatPluginError('resolved path must not be root directory');
   }
 
+  if (!fs.existsSync(outputDirectory)) {
+    fs.mkdirSync(outputDirectory, { recursive: true });
+  }
+
   if (config.clear) {
-    if (fs.existsSync(outputDirectory)) {
-      fs.rmSync(outputDirectory, { recursive: true });
+    const files = readdirRecursive(outputDirectory).filter(f => path.extname(f) === '.json');
+
+    for (let file of files) {
+      try {
+        const filepath = path.resolve(outputDirectory, file);
+        const contents = fs.readFileSync(filepath).toString();
+        new Interface(contents);
```

Comment:
This validates that the JSON file is an ABI (by throwing if it is not), correct?

Reply:
Yes.  I haven't been able to find a validator that returns true/false, so we just rely on the `Interface` initializer to fail if the passed data is not a valid ABI.

Reply:
> Yes. I haven't been able to find a validator that returns true/false, so we just rely on the `Interface` initializer to fail if the passed data is not a valid ABI.

```js
const { Interface } = require('@ethersproject/abi');
const { Signer } = require('@ethersproject/abstract-signer');
const { isAddress } = require('@ethersproject/address');
const { Contract } = require('@ethersproject/contracts');
const { LedgerSigner } = require('@ethersproject/hardware-wallets');
const { isValidName } = require('@ethersproject/hash');
const { isValidMnemonic, defaultPath } = require('@ethersproject/hdnode');
const {
  getDefaultProvider,
  JsonRpcProvider,
} = require('@ethersproject/providers');
const { Wallet } = require('@ethersproject/wallet');
const getFunctionArgs = require('./utils/getParams.js');
const formatResult = require('./utils/formatResult.js');
(async () => {
  const argv = yargs.option('address', { string: true }).argv;
  prompts.override(argv);

  const { artifact } = await prompts({
    type: 'text',
    name: 'artifact',
    message: 'Path to artifact',
    validate: fs.existsSync,
    format: (path) => JSON.parse(fs.readFileSync(path)),
    initial: 'build/contracts/YourContractABI.json',
  });
  if (artifact == undefined) {
    throw 'Undefined Artifact, Terminating';
  }
  if (!artifact.abi) {
    throw 'Invalid artifact - not of type ABI';
  }
```

This is what I use and its fairly accurate. Note the code is truncated only for the relevant part for you, you can find the rest of it located here: https://github.com/manifoldfinance/libcaller/tree/master/packages/gencall-cli

Reply:
@sambacha Seems like that code validates whether an artifact contains an ABI, but what we need here is to validate that a JSON file is a valid ABI after it has already been extracted from the artifact.  I'm fine with the `Interface` approach used in this PR - if the @ethersproject/abi packagesis updated to include an internal validator in the future, I'll update this package to make use of it.

Reply:
mmm now that I think about it, would it actually be possible to offer Truffles ABI format as an option. for output format? https://github.com/trufflesuite/truffle/tree/develop/packages/abi-utils 

Cheers,

Sam

Reply:
That's a good idea, I'll look into it.

Reply:
Seems like Hardhat ABIs are already normalized.